montrek-software/montrek
# 506 generate excels for download on parallel worker

## Development Notes

- Added `host_url` to session_data
- Separated `MailingManager` from `MailingTableManger` to avoid circular imports
- Refactored a little

## How To Test

- Prepare your DB with large data:
    - Change `N_COMPANIES=500` in `mt_effectual/scripts/fake_data/generate_fake_company_data_files.py` and run

```
python manage.py runscript mt_effectual.scripts.fake_data.generate_fake_company_data_files
```

- For a large table download a xlsx and csv:

![image](https://github.com/user-attachments/assets/874e274b-fa7c-4544-ae31-e62f902e332c)

- You should receive an email:

![image](https://github.com/user-attachments/assets/14d6a5f1-468e-454d-bca3-8d4c36265c6b)


- Check the link in the mail
